Add comprehensive README with project overview, usage, and architecture (#20)
angelocordon/kamaete: README.md

# Kamaete (構えて)

> _"To prepare" or "get ready" in Japanese_

Kamaete is a macOS developer machine bootstrapping and dotfiles management system. It provides an automated, idempotent setup that can quickly configure a new macOS machine with all essential tools, applications, and personalized configurations for development work.

## 🎯 Project Overview

**Goal:** Reproducible developer machine setup for macOS

> Kamaete enables rapid onboarding of new machines or recovery from system reinstalls with a single command. Whether you're setting up a new MacBook or rebuilding after a clean install, Kamaete ensures your development environment is configured consistently and efficiently.

Reviewer: ```suggestion
Kamaete enables rapid onboarding of new machines or recovery from system reinstalls with a single command. Whether I'm setting up a new MacBook or rebuilding after a clean install, Kamaete ensures my development environment is configured consistently and efficiently.
```
